accepted/0000-link-packages-to-source-and-build.md

```diff
@@ -0,0 +1,514 @@
+
+However, adopting something like Sigstore raises several major questions (beyond the risks detailed above):
+- What security benefits does Sigstore provide?
+- How should maintainers authenticate to sign packages? The solution should provide both high assurance in the authenticity of packages and respect maintainer privacy.
+
+#### What security benefits does Sigstore provide?
+[Sigstore](https://www.sigstore.dev/) is a collection of tools and services aimed at making it easy to use short-lived, ephemeral certificates to sign software.
+
+Sigstore has three main components: a CLI tool (Cosign), a Certificate Authority ([fulcio.sigstore.dev](https://docs.sigstore.dev/fulcio/overview/)), and a time stamping and immutable ledger service ([rekor.sigstore.dev](https://docs.sigstore.dev/rekor/overview/)).
+
+The Cosign functionality will be embedded directly in npm CLI, so we're left with the public [fulcio.sigstore.dev](https://docs.sigstore.dev/fulcio/overview/) (CA) and [rekor.sigstore.dev](https://docs.sigstore.dev/rekor/overview/) (ledger) services.
+
+##### What does [fulcio.sigstore.dev](https://docs.sigstore.dev/fulcio/overview/) give us?
+- Independent party to validate claims of the OIDC identity token which contains references back to the repo, workflow run and git SHA.
+- A supported CI/CD provider can't falsify these claims and no one else can claim to be one of these CI/CD providers.
+- The npm build script can't forge the source provenance by, for example, saying it's from some other repo or git SHA.
+- Vendor-neutral: can support any standards-compliant OIDC Identity Provider that can meet their requirements.
+
+##### What does [rekor.sigstore.dev](https://docs.sigstore.dev/rekor/overview/) give us?
+- Verifies that the signing happened during the time the certificate was valid and that the response from [Rekor](https://docs.sigstore.dev/rekor/overview/) is signed and can be verified offline.
+- Verifiable, append-only ledger, makes it hard to tamper with already signed packages.
+- Verification won't rely on [Rekor](https://docs.sigstore.dev/rekor/overview/) as we'll store detached signatures on the registry for performance and reliability but it adds an extra layer of protection in case of registry compromise.
+
+#### How should maintainers authenticate to sign packages?
+[Sigstore](https://www.sigstore.dev/) issues short-lived “disposable keys” notarized against log-ins with OIDC identities. So what OIDC identity provider (IdP) do maintainers use?
+
+This has been the subject of some controversy; for instance, in the [proposal to add Sigstore signing to RubyGems](https://github.com/rubygems/rfcs/pull/37), users worried about “vendorization” (making the repository reliant on third parties to function, e.g. GitHub, Google or Microsoft) and privacy (some maintainers are pseudonymous; Sigstore certificates include email addresses by default when signing using the [cosign](https://github.com/sigstore/cosign) CLI tool).
```

Comment:
this should be considered a big concern for npm as well (not the email part tho)

Reply:
Would npm consider being an OIDC Provider for users who do not want to rely on a third party provider? It already has unique user names, manages user identities, and has 2FA support.

Reply:
It is an option, that was brought up in https://github.com/npm/rfcs/pull/626#discussion_r940621520.

While it does provide a good user experience, the concern is that it doesn't isolate the identity provider from the package repository, so a compromise of a user account for one would compromise both.